rust/composefs/src/dumpfile.rs

```diff
@@ -445,4 +472,43 @@ mod tests {
             assert_eq!(e, e2);
         }
     }
+
+    fn parse_all(name: &str, s: &str) -> Result<()> {
+        for line in s.lines() {
+            if line.is_empty() {
+                continue;
+            }
+            let _: Entry =
+                Entry::parse(line).with_context(|| format!("Test case={name:?} line={line:?}"))?;
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_should_fail() {
+        const CASES: &[(&str, &str)] = &[
+            (
+                "long link",
+                include_str!("../../../tests/assets/should-fail-long-link.dump"),
+            ),
+            (
+                "no ftype",
+                include_str!("../../../tests/assets/should-fail-no-ftype.dump"),
```

Comment:
Note I *didn't* need to change the Rust side for this one, it was already strictly validating the mode :trophy: